smaps/src/main/webapp/js/memory-map.js

```diff
@@ -12,4 +12,118 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- */
\ No newline at end of file
+ */
+
+function drawMemoryMap() {
+  // Fetches the Json object from the MemoryMap servlet.
+  fetch('/memorymap')
+      .then((response) => {
+        return response.json();
+      })
+      .then((memoryMapJson) => {
+        // Set the rectangle size values.
+        var x = 20;   // X-coordinate of the upper-left corner of the rectangle.
+        var y = 20;   // Y-coordinate of the upper-left corner of the rectangle.
+        var w = 500;  // Width of the rectangle (pixels).
+        var h = 40;   // Height of the rectangle (pixels).
+
+        // Get the number of regions.
+        var numRegs = memoryMapJson.length;
+
+        // Get the canvas for putting the region rectangles on.
+        var c = document.getElementById('memory-map-canvas');
+
+        // Set the canvas height to be tall enough to display all regions, plus
+        // five more for a buffer.
+        c.width = 550;
+        c.height = h * (numRegs + 5);
+
+        // Draw all the region rectangles.
+        var reg = c.getContext('2d');
+        for (var i = memoryMapJson.length - 1; i >= 0; i--) {
+          // Get this region's address range and permissions.
+          var text = memoryMapJson[i][0];
+          var perms = memoryMapJson[i][1];
+
+          // Use black to draw the border of the rectangle with line width of 2.
+          reg.beginPath();
+          reg.lineWidth = '2';
+          reg.strokeStyle = 'black';
+          reg.rect(x, y, w, h);
+          reg.stroke();
+
+          // Use permissions to determine color to fill in the region rectangle.
+          if (perms == '---p') {
+            reg.fillStyle = '#BDC1C6';  // Gray, 400 intensity.
+          } else if (perms == 'rw-p') {
+            reg.fillStyle = '#78D9EC';  // Teal, 300 intensity.
+          } else if (perms == 'r-xp') {
+            reg.fillStyle = '#81C995';  // Green, 300 intensity.
+          } else if (perms == 'r--s') {
+            reg.fillStyle = '#FCAD70';  // Orange, 300 intensity.
+          } else if (perms == 'r--p') {
+            reg.fillStyle = '#C58AF9';  // Purple, 300 intensity.
+          } else if (perms == 'rw-s') {
+            reg.fillStyle = '#FF8BCB';  // Pink, 300 intensity.
+          } else if (perms == 'r-xs') {
+            reg.fillStyle = '#669DF6';  // Blue, 400 intensity.
+          }
+          reg.fill();
```

Comment:
This seems like a good section to pull out into a helper method, maybe "fillColor()"?

Reply:
On it!